subsys/bluetooth/shell/bt.c

```diff
@@ -193,26 +193,37 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
-static void print_data_set(uint8_t type, uint8_t set_value_len,
+static void print_data_hex(const uint8_t *data, uint8_t len, bool reverse)
+{
+	shell_fprintf(ctx_shell, SHELL_INFO, "0x");
+	for (uint8_t i = 0; i < len; i++) {
+		shell_fprintf(ctx_shell, SHELL_INFO, "%02x", data[reverse ? len-1-i : i]);
+	}
+}
+
+static void print_data_set(uint8_t type, uint8_t initial_value_len, uint8_t set_value_len,
 			   const uint8_t *scan_data, uint8_t scan_data_len)
 {
-	uint8_t min_value_len = MIN(set_value_len, scan_data_len);
+	uint8_t idx = 0;
 
 	shell_fprintf(ctx_shell, SHELL_INFO, "%*sType 0x%02x: ",
-		      strlen(scan_response_label), "",
-		      type);
-	for (uint8_t i = 0U; i < (scan_data_len / min_value_len); i++) {
-		if (i > 0L) {
-			shell_fprintf(ctx_shell, SHELL_INFO, ", ");
-		}
+			  strlen(scan_response_label), "",
+			  type);
 
-		shell_fprintf(ctx_shell, SHELL_INFO, "0x");
-		for (uint8_t j = 0U; j < min_value_len; j++) {
-			shell_fprintf(ctx_shell, SHELL_INFO, "%02x",
-				      scan_data[i * min_value_len + j]);
-		}
+	if (initial_value_len > 0) {
+		print_data_hex(scan_data, initial_value_len, true);
+		shell_fprintf(ctx_shell, SHELL_INFO, ",");
+		idx = initial_value_len;
 	}
 
+	do {
+		if (idx > initial_value_len) {
+			shell_fprintf(ctx_shell, SHELL_INFO, ",");
+		}
+		print_data_hex(&scan_data[idx], set_value_len, true);
```

Comment:
`print_data_hex` seems to always be called with `reverse = true` which seems like a bug. 

If this `reverse` part is the handle endianess, then I would suggest to instead do a switch on the data_len and use the `sys_leXX_to_cpu` for all values < 64 bits. 

Any value > 64 bits in Bluetooth are very likely to be arrays (with the exception of 128-bit UUIDs). We may, however, want to use `bt_uuid_to_str` to handle UUIDs.

Reply:
That is a fair point - It makes sense to print as MSB first